File: 01_projects/code/python-playwright/cbaps-dawms/robust/dawms/pages/portal_home_page.py
from playwright.sync_api import Page
from loguru import logger

class PortalHomePage:
    def __init__(self, page: Page, pw):
        self.page = page
        self.pw = pw
        self.dawms_link = page.locator("a:has-text('DAWMS')")
    
    def navigate_to_portal(self, url: str):
        self.pw.navigate_to(url)
        self.pw.wait_for_selector("a:has-text('DAWMS')")
    
    def open_dawms(self):
        from .dawms_dashboard_page import DAWMSDashboardPage
        self.pw.click_element(self.dawms_link)
        logger.info("➡️ Opening DAWMS")
        return DAWMSDashboardPage(self.page, self.pw)

File: 01_projects/code/typescript-playwright/cbaps-dawms/dawms/pages/PortalHomePage.ts
/**
 * PortalHomePage: Portal entry point for DAWMS
 */

import { Page, Locator } from '@playwright/test';
import { PlaywrightManager } from '../library/PlaywrightManager';
import { DAWMSDashboardPage } from './DAWMSDashboardPage';

export class PortalHomePage {
  readonly page: Page;
  readonly pwManager: PlaywrightManager;

  readonly dawmsLink: Locator;
  readonly headerLogo: Locator;

  constructor(page: Page, pwManager: PlaywrightManager) {
    this.page = page;
    this.pwManager = pwManager;

    this.dawmsLink = page.locator("a:has-text('DAWMS')");
    this.headerLogo = page.locator('.portal-logo');

    this.pwManager.waitVisible('text=Application Portal', 'Portal page loaded');
    console.log('✓ Portal Home Page initialized');
  }

  async navigateToPortal(url: string): Promise<this> {
    await this.pwManager.navigateTo(url);
    await this.pwManager.waitVisible('text=Application Portal', 'Waiting for portal');
    return this;
  }

  async openDAWMS(): Promise<DAWMSDashboardPage> {
    await this.pwManager.click(this.dawmsLink);
    console.log('✓ Clicked DAWMS link');
    return new DAWMSDashboardPage(this.page, this.pwManager);
  }

  async validatePortalLoaded(): Promise<boolean> {
    const isDawmsVisible = await this.pwManager.isVisible(this.dawmsLink);
    const isLogoVisible = await this.pwManager.isVisible(this.headerLogo);
    return isDawmsVisible && isLogoVisible;
  }

  async getTitle(): Promise<string> {
    return await this.pwManager.getTitle();
  }
}
